vtex/faststore
# feat: Added ability to set target="_blank" to FS Core Hero and Banner

## What's the purpose of this pull request?
Currently there are a few FastStore core components containing links that can't be authored to open the link in a new window. See GH issue: https://github.com/vtex/faststore/issues/2097

## How it works?
I updated `sections.json` to allow authoring in the admin panel and added custom components to the overrides for `BannerText` and `Hero` to pass the `target="_blank"` attribute to the links they contain.

I had trouble deciding whether to add this logic in FastStore Core vs FastStore UI components, and eventually landed on ~~Core because it seemed preferable to making the UI components less generic. However I think there's also an argument for adding the logic to FastStore UI molecules and organisms where it's applicable.~~ FastStore UI. I'm curious about whether this makes sense to the FS team.

## How to test it?
- Build locally and symlink `@faststore/ui` in your local store to `packages/core` (or use the [CodeSandbox deploy](https://ci.codesandbox.io/status/vtex/faststore/pr/2108/builds/438859) Local Install Instructions)
- Sync with your workspace CMS
- Author a test `BannerText` or `Hero` to confirm that there is a checkbox labeled "Open link in new window?" is present, check it ✅ 
- Publish the test page/component and confirm that the link opens in a new window

### Starters Deploy Preview
https://starter-git-fork-katrillion-target-blank-test-faststore.vercel.app/
